From 8245477454f8ae23e75a2d9aa573d8895bdd5ae3 Mon Sep 17 00:00:00 2001
Message-Id: <8245477454f8ae23e75a2d9aa573d8895bdd5ae3.1297440281.git.jdenemar@redhat.com>
From: Gui Jianfeng <guijianfeng@cn.fujitsu.com>
Date: Tue, 8 Feb 2011 15:00:24 +0800
Subject: [PATCH] LXC: LXC Blkio weight configuration support.

https://bugzilla.redhat.com/show_bug.cgi?id=632492

LXC Blkio weight configuration support.

Reviewed-by: "Nikunj A. Dadhania" <nikunj@linux.vnet.ibm.com>
Signed-off-by: Gui Jianfeng <guijianfeng@cn.fujitsu.com>
(cherry picked from commit 76d99efc441407ce1b8e53eab4eef38572e603b3)
---
 src/lxc/lxc_controller.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/src/lxc/lxc_controller.c b/src/lxc/lxc_controller.c
index af0b70c..0db6673 100644
--- a/src/lxc/lxc_controller.c
+++ b/src/lxc/lxc_controller.c
@@ -104,6 +104,16 @@ static int lxcSetContainerResources(virDomainDefPtr def)
         goto cleanup;
     }
 
+    if (def->blkio.weight) {
+        rc = virCgroupSetBlkioWeight(cgroup, def->blkio.weight);
+        if (rc != 0) {
+            virReportSystemError(-rc,
+                                 _("Unable to set Blkio weight for domain %s"),
+                                 def->name);
+            goto cleanup;
+        }
+    }
+
     rc = virCgroupSetMemory(cgroup, def->mem.max_balloon);
     if (rc != 0) {
         virReportSystemError(-rc,
-- 
1.7.4

